charts/seed-monitoring/charts/core/charts/prometheus/rules/node-exporter.rules.yaml

```diff
@@ -1,6 +1,18 @@
 groups:
 - name: node-exporter.rules
   rules:
+  - alert: NodeKubeletDown
+    expr: count_over_time(sum by (node, type)(last_over_time(kube_node_status_condition{condition="Ready",status!="true"}[1m]) == 1)[10m:1m])
```

Comment:
Is this the last version of the expression? The last time we extensively discussed the expression and came to the conclusion that `last_over_time` for the last minute is not really meaningful.

Reply:
The expression is:
`expr: count_over_time((sum by (node)(kube_node_status_condition{condition="Ready",status!="true"}) == 1)[10m:]) == 10`

`kube_node_status_condition{condition="Ready",status!="true"} == 1`: prints a table of nodes where the status is not true (printed are nodes with status false or unknown) (`== 1` means that the lines on the table satisfy this expression)

`((kube_node_status_condition{condition="Ready",status!="true"}) == 1)[10m:]`: For EACH data point, performs the above query every evaluation interval for the past 10 minute window. Here, we have `evaluation_interval: 1m`, so this would check once every minute for the past 10 minutes. Because `== 1` exists, this will only show lines where this expression was true. So if we have 10 elements, the condition was true for 10 minutes.

`(sum by (node)(kube_node_status_condition{condition="Ready",status!="true"}) == 1)[10m:]`: `(sum by (node)` groups the results by node.

`count_over_time((sum by (node)(kube_node_status_condition{condition="Ready",status!="true"}) == 1)[10m:]) == 10`: `count_over_time` counts how many times the condition is true (instead of a 1 at every evaluation interval, this counts how many times the query returns 1). The `== 10` part only considers nodes that have had this condition true for every minute in the past 10 minute evaluation time.

This gives the full expression above.